tests/parsing/FilterTest.py

```diff
@@ -0,0 +1,128 @@
+        with bear_test_module():
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '-B',
+                '--filter-by-language', 'java',
+                '--filter-by', 'can_detect', 'formatting')
+            self.assertEqual(retval, 0)
+            # 1 bear plus 1 line holding the closing colour escape sequence.
+            self.assertEqual(len(stdout.strip().splitlines()), 2)
+
+    def test_filter_by_can_detect_syntax(self):
+        with bear_test_module():
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '-B', '--filter-by',
+                'can_detect', 'syntax')
+            self.assertEqual(retval, 0)
+            # 2 bears plus 1 line holding the closing colour escape sequence.
+            self.assertEqual(len(stdout.strip().splitlines()), 3)
+
+    def test_filter_by_can_detect_security(self):
+        with bear_test_module():
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '-B', '--filter-by',
+                'can_detect', 'security')
+            self.assertEqual(retval, 0)
+            # 1 bear plus 1 line holding the closing colour escape sequence.
+            self.assertEqual(len(stdout.strip().splitlines()), 2)
+
+    def test_filter_by_can_detect_spelling(self):
+        with bear_test_module():
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '-B', '--filter-by',
+                'can_detect', 'spelling')
+            self.assertEqual(retval, 0)
+            # 1 line holding the closing colour escape sequence.
+            self.assertEqual(len(stdout.strip().splitlines()), 1)
+            self.assertIn('No bears to show.', stdout)
+
+    def test_filter_by_can_fix_syntax(self):
+        with bear_test_module():
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '-B', '--filter-by',
+                'can_fix', 'syntax')
+            self.assertEqual(retval, 0)
+            # 1 line holding the closing colour escape sequence.
+            self.assertEqual(len(stdout.strip().splitlines()), 1)
+            self.assertIn('No bears to show.', stdout)
+
+    def test_filter_by_can_fix_redundancy(self):
+        with bear_test_module():
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '-B', '--filter-by',
+                'can_fix', 'redundancy')
+            self.assertEqual(retval, 0)
+            # 1 bear plus 1 line holding the closing colour escape sequence.
+            self.assertEqual(len(stdout.strip().splitlines()), 2)
+
+    def test_filter_by_unknown(self):
+        with bear_test_module():
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '-B', '--filter-by', 'unknown', 'arg1')
+            self.assertEqual(retval, 2)
+            self.assertEqual(
+                stdout, FilterHelper.get_help_message('unknown') + '\n')
```

Comment:
I would rather write the messages out, testing by expected output, not by expected algorithm behaviour ;)

(applies everywhere in this test ;) )